rfc/organizing-work.md

```diff
@@ -0,0 +1,58 @@
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Out of scope
+
+* The tools or locations used to document work items. i.e. Github issues, Jira, Notion etc.
+* Roadmapping/general high level direction setting.
+
+## Pipeline
+
+Generally a work item would move from inception to completion through a pipeline in the following order:
+
+1. **Heap** - anyone can pile stuff onto the heap. It's just a place to track things that we may want to work on in the future. It can be of arbitrary size and detail.
+
+    Items that are of ambiguous _certainty_ SHOULD be accompanied by an [RFC](https://github.com/web3-storage/rfc) - a PR that allows others to comment in an open and structured way on the item. For example, an RFC may be a new idea or a way of achieving a goal that could be done in multiple ways. An RFC allows others to validate the idea, question or assumption to determine with more certainty if it should be worked on.
+    
+    All members SHOULD read RFC documents and provide feedback to allow the author to determine validity of the work item and adjust or clean up if necessary.
+
+2. **Stack** - prioritized work items. _Unlike_ a conventional stack, items are added at a position that denotes their relative priority, however, as per a conventional stack, items are usually "popped" from the top.
+
+    Items SHOULD be moved from the heap to the stack as part of a regular lightweight planning exercise that SHOULD include members of the group that will be working on the items. At this point a short description that details the _product benefits_ and _desired end state/outcome(s)_ of the system MUST be added if not already present.
+
+    Items on the stack SHOULD be prioritized according to the general roadmap as well as other factors, such as, but limited to, how on fire things are. Items in the stack may be re-prioritized during group planning.
+    
+    Multiple items in the heap MAY result in a _single_ item on the stack.
+
+    Larger items whose outputs are an implementation of some kind MUST also result in a PR to the [specs](https://github.com/web3-storage/specs/) repo.
+
+3. **Phase** - a short period of time where items on the stack are worked on. It focuses group effort on a particular theme and is analogous to a "sprint" in Scrum.
+
+    A phase MUST be created by group members who will be working on the items. Members decide on a theme and select a set of items from the top of the stack to work on. At this time a size estimate SHOULD be calculated per item to determine whether enough time exists to achieve the task within the phase period.
```

Comment:
Do you imagine multiple Phases in progress at the same time across the organization? It feels like that will inevitably be needed as the team grows and work streams diverge more, but maybe we're not there yet? Do you have any thoughts on how many people would be involved in a single phase or is that out of scope for this doc?

Reply:
I would also prefer to work in thematic way as much as possible. I think it can be mostly achieved if we have smaller phases, at least for a team of our size.

However, at the bare minimal to accomodate team growth, I think we should commit to:
- have non silos where one single member takes a thematic phase
- not have members spread into more than one thematic phase (unless of course it finishes half way through phase)

Reply:
If we going to stick to terminology borrowed from computer science, this should be called threads 😅

Jokes aside I do agree with @vasco-santos single person per theme / thread creates silos echo chambers and demotivation. I would also argue that goal should be more about a checkpoint and reevaluation than it is to drain an allocated stack. In our prior retrospectives we keep pushing timelines to drain the stack, it would be more rewarding to just call it re-evaluate and start another phase / thread.